src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -53,15 +65,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
     UserService userService = UserServiceFactory.getUserService();
     if (!userService.isUserLoggedIn()) {
       response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
       return;
     }
 
+    addEvent(request, response, userService);
+
     // Redirect back to the HTML page.
-    response.sendRedirect("/index.html");
+    if (response.getStatus() == HttpServletResponse.SC_OK) {
+      response.sendRedirect("/index.html");
+    }
```

Comment:
Maybe you can return a bool from addEvent which indicates whether it was successful or not and don't use the status here as an indicator for this.
Maybe also add a TODO to make proper error handling, because currently I assume that the user don't see a proper error message when an error occurred :)